torch/csrc/cuda/python_nccl.cpp

```diff
@@ -294,3 +295,21 @@ PyObject * THCPModule_nccl_reduce_scatter(PyObject *self, PyObject *args) {
   Py_RETURN_NONE;
   END_HANDLE_TH_ERRORS
 }
+
+static std::vector<at::Tensor> extract_tensors(PyObject* obj) {
+  auto seq = THPObjectPtr(PySequence_Fast(obj, "expected a sequence"));
+  if (!seq) throw python_error();
+
+  std::vector<at::Tensor> list;
+  Py_ssize_t length = PySequence_Fast_GET_SIZE(seq.get());
+  for (Py_ssize_t i = 0; i < length; i++) {
+    PyObject* item = PySequence_Fast_GET_ITEM(seq.get(), i);
+    if (THPVariable_Check(item)) {
+      auto var = (THPVariable*) item;
+      list.emplace_back(var->cdata.data());
```

Comment:
Should we check that they don't require grad? It might give a false impression that this is differentiable